zerver/lib/string_validation.py

```diff
@@ -0,0 +1,93 @@
+import unicodedata
+
+from django.utils.translation import gettext as _
+
+from zerver.lib.exceptions import JsonableError
+
+unicode_non_chars = [
+    "\uFDD0",
+    "\uFDD1",
+    "\uFDD2",
+    "\uFDD3",
+    "\uFDD4",
+    "\uFDD5",
+    "\uFDD6",
+    "\uFDD7",
+    "\uFDD8",
+    "\uFDD9",
+    "\uFDDA",
+    "\uFDDB",
+    "\uFDDC",
+    "\uFDDD",
+    "\uFDDE",
+    "\uFDDF",
+    "\uFDE0",
+    "\uFDE1",
+    "\uFDE2",
+    "\uFDE3",
+    "\uFDE4",
+    "\uFDE5",
+    "\uFDE6",
+    "\uFDE7",
+    "\uFDE8",
+    "\uFDE9",
+    "\uFDEA",
+    "\uFDEB",
+    "\uFDEC",
+    "\uFDED",
+    "\uFDEE",
+    "\uFDEF",
+    "\uFFFE",
+    "\uFFFF",
+    "\U0001FFFE",
+    "\U0001FFFF",
+    "\U0002FFFE",
+    "\U0002FFFF",
+    "\U0003FFFE",
+    "\U0003FFFF",
+    "\U0004FFFE",
+    "\U0004FFFF",
+    "\U0005FFFE",
+    "\U0005FFFF",
+    "\U0006FFFE",
+    "\U0006FFFF",
+    "\U0007FFFE",
+    "\U0007FFFF",
+    "\U0008FFFE",
+    "\U0008FFFF",
+    "\U0009FFFE",
+    "\U0009FFFF",
+    "\U0010FFFE",
+    "\U0010FFFF",
+    "\U000AFFFE",
+    "\U000AFFFF",
+    "\U000BFFFE",
+    "\U000BFFFF",
+    "\U000CFFFE",
+    "\U000CFFFF",
+    "\U000DFFFE",
+    "\U000DFFFF",
+    "\U000EFFFE",
+    "\U000EFFFF",
+    "\U000FFFFE",
+    "\U000FFFFF",
+]
```

Comment:
This is very verbose -- since there's a simple pattern behind it, we can probably make it more understandable and concise.  And magic constants like this are the things that benefit greatly from a comment pointing to their source -- otherwise, they just leave me wondering "why is it _this_ set of of things?"

```suggestion
# There are 66 Unicode non-characters; see 
# https://www.unicode.org/faq/private_use.html#nonchar4
unicode_non_chars = [
    chr(x)
    for x in list(range(0xFDD0, 0xFDF0)) # FDD0 through FDEF, inclusive
    + list(range(0xFFFE, 0x110000, 0x10000))  # 0xFFFE, 0x1FFFE, ... 0x10FFFE inclusive
    + list(range(0xFFFF, 0x110000, 0x10000))  # 0xFFFF, 0x1FFFF, ... 0x10FFFF inclusive
]